src/data/fintype/basic.lean

```diff
@@ -1293,8 +1293,8 @@ by { ext, simp [finset.mem_powerset_len] }
   fintype.card {s : finset α // s.card = k} = nat.choose (fintype.card α) k :=
 by simp [fintype.subtype_card, finset.card_univ]
 
-@[simp] lemma set.to_finset_univ [fintype α] :
-  (set.univ : set α).to_finset = finset.univ :=
+@[simp] lemma set.to_finset_univ [hα : fintype α] [hu : fintype (set.univ : set α)] :
```

Comment:
Do you want to infer `hα` and `hu` using TC? Or would it be better to use unification, and hence use `{}`-binders?

Reply:
That's a better idea. Now there's the `simp` version that uses implicit arguments along with the original one (still useful if you want to do reverse rewrites.)